chronos/apps/http/httpserver.nim

```diff
@@ -450,47 +469,30 @@ proc getRequest(conn: HttpConnectionRef): Future[HttpRequestRef] {.async.} =
   except AsyncStreamLimitError:
     raiseHttpCriticalError("Maximum size of request headers reached", Http431)
 
+proc init*(value: var HttpConnection, server: HttpServerRef,
+           transp: StreamTransport) =
+  value.server = server
```

Comment:
a way to write an `init` and guarantee that all fields are reset is `value = HttpConnection(a: b, ...)`, if that is the intent